eng/scripts/TypeSpec-Validation.ps1

```diff
@@ -1,16 +1,35 @@
 [CmdletBinding()]
 param (
   [switch]$CheckAll = $false,
+  [int]$Shard = 0,
+  [int]$TotalShards = 1,
   [switch]$GitClean = $false,
   [switch]$DryRun = $false,
   [string]$BaseCommitish = "HEAD^",
   [string]$TargetCommitish = "HEAD"
 )
 
+if ($TotalShards -gt 0 -and $Shard -ge $TotalShards) { 
+  throw "Shard ($Shard) must be less than TotalShards ($TotalShards)"
+}
+
 . $PSScriptRoot/Logging-Functions.ps1
 . $PSScriptRoot/Suppressions-Functions.ps1
+. $PSScriptRoot/Array-Functions.ps1
+
+$typespecFolders, $checkedAll = &"$PSScriptRoot/Get-TypeSpec-Folders.ps1" `
+  -BaseCommitish:$BaseCommitish `
+  -TargetCommitish:$TargetCommitish `
+  -CheckAll:$CheckAll
 
-$typespecFolders, $checkedAll = &"$PSScriptRoot/Get-TypeSpec-Folders.ps1" -BaseCommitish:$BaseCommitish -TargetCommitish:$TargetCommitish -CheckAll:$CheckAll
+if ($TotalShards -gt 1) { 
```

Comment:
```suggestion
if ($TotalShards -gt 1 -and $TotalShards -le $typespecFolders.Count) { 
```
Prevents exception, if the request number of shards is more than the number of specs (unlikely, but possible)